backend/routes/conversation.py

```
@@ -0,0 +1,142 @@
+    history_messages.append({"role": "user", "content": request.user_message})
+
+    try:
+        assistant_text = backend.generate_with_history(
+            history=history_messages,
+            system=system_prompt,
+            max_tokens=512,
+            temperature=0.7,
+        )
+    except Exception as e:
+        logger.error("LLM call failed: %s", e)
+        raise HTTPException(status_code=502, detail=f"LLM request failed: {str(e)}")
```

Ruff (0.15.21)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)